src/libstore/s3_binary_cache_store.go

```diff
@@ -0,0 +1,101 @@
+package libstore
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"net/url"
+	"github.com/minio/minio-go/v7"
+	"github.com/minio/minio-go/v7/pkg/credentials"
+	"log"
+)
+
+// S3BinaryCacheStore ...
+type S3Store struct {
+	Profile string // The name of the AWS configuration profile to use.
+	Region string // "aws-region"
+	Scheme string //  "The scheme to use for S3 requests, https by default."
+	// narinfoCompression string // "compression method for .narinfo files"
+	// multipartUpload bool
+	// bufferSize uint64
+ // do we need to put endpoint in here
+ // anything that is needed in minio, no need here. Check.
+	bucketName string
+  client *minio.Client
+}
+
+// SDK should be in environment
+func (s *S3Store) getUri() string {
+  return "s3://" + s.bucketName
+}
+
+func (s *S3Store) FileExists(ctx context.Context, path string) (bool, error) {
+
+	return true
+}
+
+// TODO: Implement GetFile for S3
+func (s *S3Store) GetFile(ctx context.Context, path string) (io.ReadCloser, error) {
+	// extend the struct to have the client
+	object, err := s.client.GetObject(s.bucketName, path, minio.GetObjectOptions{})
+	if err != nil {
+			fmt.Println(err)
+			return
+	}
```

Comment:
the client should return the object here since it's a reader.

Reply:
do you mean if it is error, then we have to return `error` or `io.ReaderCloser`?

Reply:
if the error is not nil then it `return nil, err`. Otherwise it would `return object, nil`.